sonarqube/content.md

```diff
@@ -30,68 +45,145 @@ $ mvn sonar:sonar -Dsonar.host.url=http://$(boot2docker ip):9000
 
-The production database is configured with the following SonarQube properties used as environment variables: `sonar.jdbc.username`, `sonar.jdbc.password` and `sonar.jdbc.url`.
+5. By default, SonarQube is configured to run on any computer with a simple Java JRE.
+
+For better performance, the first thing to do when installing a production instance is to use a Java JDK and activate the server mode by uncommenting/setting the following line in $SQ_HOME/conf/sonar.properties:
+```
+sonar.web.javaOpts=-server
+```
+To change the Java JVM used by SonarQube, simply edit $SQ_HOME/conf/wrapper.conf and update the following line:
+```
+wrapper.java.command=/path/to/my/jdk/bin/java
+```
 
+6. Run the image:
 ```console
 $ docker run -d --name sonarqube \
     -p 9000:9000 \
-    -e sonar.jdbc.username=sonar \
-    -e sonar.jdbc.password=sonar \
-    -e sonar.jdbc.url=jdbc:postgresql://localhost/sonar \
+    -v $SQ_HOME/conf:/opt/sonarqube/conf \
+    -v $SQ_HOME/extensions:/opt/sonarqube/extensions \
+    -v $SQ_HOME/logs:/opt/sonarqube/logs \
+    -v $SQ_HOME/data:/opt/sonarqube/data \
     sonarqube
 ```
 
-Use of the environment variables `SONARQUBE_JDBC_USERNAME`, `SONARQUBE_JDBC_PASSWORD` and `SONARQUBE_JDBC_URL` is deprecated, and will stop working in future releases.
+## Upgrade SonarQube
+Follow these steps to upgrade SonarQube:
 
-More recipes can be found [here](https://github.com/SonarSource/docker-sonarqube/blob/master/recipes.md).
+1. Create a new `sonarqube_home` folder and backup your old folder.
 
-You can pass `sonar.` configuration properties as Docker environment variables, as demonstrated in the example above for database configuration.
+2. Update the var env `$SQ_HOME` pointing to it. Backup the old one:
+```console
+$ mkdir /path/to/your/filesystem/sq_home81
+$ export SQ_HOME=/path/to/your/filesystem/sq_home81
+```
+
+3. Initialize the new `SQ_HOME` with `--init`:
+```console
+$ docker run --rm \
+    -v $SQ_HOME/conf:/opt/sonarqube/conf \
+    -v $SQ_HOME/extensions:/opt/sonarqube/extensions \
+    -v $SQ_HOME/data:/opt/sonarqube/data \
+    sonarqube --init  
+```
+
+4. Follow the normal upgrade procedure https://docs.sonarqube.org/latest/setup/upgrading/
+  - Manually install the non-default plugins
+  - Update the contents of sonar.properties and wrapper.conf
+  - etc.  
+  
+5. Stop and remove the sonarqube container (restart is not enough as the env variable are not evaluated only during the first run, not during the restart):
+```console
+$ docker stop sonarqube
+$ docker rm sonarqube
+```  
+
+6. Run docker:
+```console
+$ docker run -d --name sq -p 9000:9000 \
+    -v $SQ_HOME/conf:/opt/sonarqube/conf \
+    -v $SQ_HOME/extensions:/opt/sonarqube/extensions \
+    -v $SQ_HOME/logs:/opt/sonarqube/logs \
+    -v $SQ_HOME/data:/opt/sonarqube/data \
+    sonarqube
+```
 
-### Option 2: Use bind-mounted persistent volumes
+Other option : use a symlink instead of an env. variable
```

Comment:
I don't know how to interpret this. It's kind of lost at the end.

Reply:
kind of. The idea was to say that we could use symlink instead of env. var, but it only apply to linux. But as a linux admin, i would rather use symlink than env var. We specified the doc with env variable instead because it work on linux/mac/windows

Reply:
Would it make more sense below step 2 to say something like,  **Note:** If you're using Linux, you can use a symlink instead of an environment variable.

Reply:
to me yes :)